IndexOutOfBoundsException when using createFor(grid) before grid added to UI

### Describe the bug

I got a strange error today with the GridExporter addon.
Not sure if it is comming from the GridExporter or from the GridHelper library, or if it's only related to the background thread (detached UI).

GridExporter v2.5.1
GridHelper v1.4.2
Vaadin 24.6.7

**Code which throws the error:**
```
 ui.access(()
    -> {
    Grid grid= new Grid<>();
    // Gives a strange array out of bounds error in setFooterToolbar
    **GridExporter<ImportMarginSheetRow> gExporter= GridExporter.createFor(grid);**
    validateStep.add(grid);
}
```

**Code which works**
```
 ui.access(()
    -> {
    Grid grid= new Grid<>();
    // Add grid first, before attaching the grid exporter to preven tOutOfBound exception
    validateStep.add(grid);
    GridExporter<ImportMarginSheetRow> gExporter= GridExporter.createFor(grid);
}
```

**Throws:**
```
Caused by: java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.base/java.util.Objects.checkIndex(Objects.java:361)
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at java.base/java.util.Collections$UnmodifiableList.get(Collections.java:1347)
	at com.flowingcode.vaadin.addons.gridhelpers.FooterToolbarGridHelper.setFooterToolbar(FooterToolbarGridHelper.java:50)
	at com.flowingcode.vaadin.addons.gridhelpers.GridHelper.addToolbarFooter(GridHelper.java:352)
	at com.flowingcode.vaadin.addons.gridexporter.GridExporter.lambda$createFor$afa35fd0$1(GridExporter.java:197)
	at com.vaadin.flow.dom.Element$1.execute(Element.java:1254)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at com.vaadin.flow.internal.StateNode.fireAttachListeners(StateNode.java:931)
	at com.vaadin.flow.internal.StateNode.onAttach(StateNode.java:335)
	at com.vaadin.flow.internal.StateNode.setParent(StateNode.java:287)
	at com.vaadin.flow.internal.nodefeature.NodeFeature.attachPotentialChild(NodeFeature.java:80)
	at com.vaadin.flow.internal.nodefeature.StateNodeNodeList.add(StateNodeNodeList.java:55)
	at com.vaadin.flow.internal.nodefeature.ElementChildrenList.add(ElementChildrenList.java:44)
	at com.vaadin.flow.dom.impl.AbstractNodeStateProvider.insertChild(AbstractNodeStateProvider.java:107)
	at com.vaadin.flow.dom.Node.insertChild(Node.java:391)
	at com.vaadin.flow.dom.Node.appendChild(Node.java:163)
	at com.vaadin.flow.dom.Node.appendChild(Node.java:147)
	at com.vaadin.flow.component.HasComponents.lambda$add$1(HasComponents.java:75)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at com.vaadin.flow.component.HasComponents.add(HasComponents.java:75)
	at com.vaadin.flow.component.HasComponents.add(HasComponents.java:58)
	at xx.xxx.xxx.xxx.importdata.ImportSubmailingWizard$ValidateRunner.lambda$run$1e6668b$1(ImportSubmailingWizard.java:424)
	at com.vaadin.flow.component.UI.accessSynchronously(UI.java:498)
	at com.vaadin.flow.component.UI$2.execute(UI.java:573)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java)
```

### Add-on Version

2.5.1

### Vaadin Version

24.6.7

### Additional information

_No response_

--- a/src/main/java/com/flowingcode/vaadin/addons/gridhelpers/FooterToolbarGridHelper.java
+++ b/src/main/java/com/flowingcode/vaadin/addons/gridhelpers/FooterToolbarGridHelper.java
@@ -2,7 +2,7 @@
  * #%L
  * Grid Helpers Add-on
  * %%
- * Copyright (C) 2022 - 2024 Flowing Code
+ * Copyright (C) 2022 - 2025 Flowing Code
  * %%
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -36,6 +36,9 @@ class FooterToolbarGridHelper implements Serializable {
 
   public void setFooterToolbar(Component toolBar) {
     Grid<?> grid = helper.getGrid();
+    if (grid.getColumns().isEmpty()) {
+      throw new IllegalStateException("Cannot set footer toolbar: Grid columns have not been configured.");
+    }
     if (grid.getFooterRows().isEmpty()) {
       // create a fake footer and hide it (workaround:
       // https://github.com/vaadin/flow-components/issues/1558#issuecomment-987783794)

--- a/src/main/java/com/flowingcode/vaadin/addons/gridhelpers/GridHelper.java
+++ b/src/main/java/com/flowingcode/vaadin/addons/gridhelpers/GridHelper.java
@@ -2,7 +2,7 @@
  * #%L
  * Grid Helpers Add-on
  * %%
- * Copyright (C) 2022 - 2024 Flowing Code
+ * Copyright (C) 2022 - 2025 Flowing Code
  * %%
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -348,6 +348,16 @@ public final class GridHelper<T> implements Serializable {
 
   private final FooterToolbarGridHelper footerToolbar = new FooterToolbarGridHelper(this);
 
+  /**
+   * Adds a toolbar component to the footer of the grid.
+   * <p>
+   * Note: The grid must have its columns configured before calling this method.
+   * Otherwise, an {@link IllegalStateException} will be thrown.
+   *
+   * @param grid the grid to add the toolbar to
+   * @param toolBar the toolbar component to add
+   * @throws IllegalStateException if the grid columns have not been configured
+   */
   public static void addToolbarFooter(Grid<?> grid, Component toolBar) {
     getHelper(grid).footerToolbar.setFooterToolbar(toolBar);
   }

--- a/src/test/java/com/flowingcode/vaadin/addons/gridhelpers/test/FooterToolbarTest.java
+++ b/src/test/java/com/flowingcode/vaadin/addons/gridhelpers/test/FooterToolbarTest.java
@@ -2,7 +2,7 @@
  * #%L
  * Grid Helpers Add-on
  * %%
- * Copyright (C) 2022 - 2024 Flowing Code
+ * Copyright (C) 2022 - 2025 Flowing Code
  * %%
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -22,28 +22,24 @@ package com.flowingcode.vaadin.addons.gridhelpers.test;
 import com.flowingcode.vaadin.addons.gridhelpers.GridHelper;
 import com.vaadin.flow.component.grid.Grid;
 import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
-import org.junit.Before;
 import org.junit.Test;
-import org.junit.Test.None;
 
 public class FooterToolbarTest {
-  
-  private Grid<Bean> grid;
 
-  private HorizontalLayout toolbarFooter;
-  
-  private class Bean {}
-  
-  @Before
-  public void before() {
-    grid = new Grid<>(Bean.class, false);
+  private static class Bean {}
+
+  @Test
+  public void addToolbarFooter() {
+    Grid<Bean> grid = new Grid<>(Bean.class, false);
     grid.addColumn(x -> x).setHeader("Header");
-    toolbarFooter = new HorizontalLayout();    
-  }
-  
-  @Test(expected = None.class)
-  public void addToolbarFooter_toolbarFooterIsShown() {
+    var toolbarFooter = new HorizontalLayout();
     GridHelper.addToolbarFooter(grid, toolbarFooter);
   }
 
+  @Test(expected = IllegalStateException.class)
+  public void testSetFooterToolbarBeforeColumnsConfiguredThrowsException() {
+    Grid<Bean> grid = new Grid<>(Bean.class, false);
+    var toolbarFooter = new HorizontalLayout();
+    GridHelper.addToolbarFooter(grid, toolbarFooter);
+  }
 }

